src/hipscat/catalog/partition_info.py

```diff
@@ -44,10 +50,54 @@ def write_to_file(self, partition_info_file: FilePointer):
         """
         file_io.write_dataframe_to_csv(self.as_dataframe(), partition_info_file, index=False)
 
+    def write_to_metadata_files(self, catalog_path: FilePointer, storage_options: dict = None):
+        """Generate parquet metadata, using the known partitions.
+
+        Args:
+            catalog_path (str): base path for the catalog
+            storage_options: dictionary that contains abstract filesystem credentials
+        """
+        batches = [
+            pa.RecordBatch.from_arrays(
+                [[pixel.order], [pixel.dir], [pixel.pixel]],
+                names=[
+                    self.METADATA_ORDER_COLUMN_NAME,
+                    self.METADATA_DIR_COLUMN_NAME,
+                    self.METADATA_PIXEL_COLUMN_NAME,
+                ],
+            )
+            for pixel in self.get_healpix_pixels()
+        ]
+
+        write_parquet_metadata_for_batches(batches, catalog_path, storage_options)
+
+    @classmethod
+    def read_from_file(cls, metadata_file: FilePointer, storage_options: dict = None):
```

Comment:
Could we add the return type for this method and the data types for the arguments? Also for `read_from_csv`, `row_group_stat_single_value`, `read_row_group_fragments` and `is_metadata_valid` and `dir` - both in the stubs and in the docstrings?

Reply:
Done!